src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -102,7 +102,12 @@ CHIP_ERROR ConfigurationManagerImpl::_Init()
 
 CHIP_ERROR ConfigurationManagerImpl::_GetPrimaryWiFiMACAddress(uint8_t * buf)
 {
-    return esp_wifi_get_mac(WIFI_IF_STA, buf);
+    wifi_mode_t mode;
+    esp_wifi_get_mode(&mode);
+    if ((mode == WIFI_MODE_AP) || (mode == WIFI_MODE_APSTA))
+        return esp_wifi_get_mac(WIFI_IF_AP, buf);
```

Comment:
I believe esp_wifi_get_mac returns esp_err_t instead of CHIP_ERROR

could we fix the return code now that we are touching this part of the code?

Reply:
fixed